File: cs214/projects/proj10/ruby/birds.rb
# birds.rb | Tests out the menagerie of bird classes.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Advait Scaria
# Date: 04/23/2019
######################################################

require './Bird.rb'
require './FlyingBird.rb'
require './WalkingBird.rb'
require './Duck.rb'
require './Goose.rb'
require './Owl.rb'
require './Penguin.rb'
require './Ostrich.rb'
require './Kiwi.rb'

puts "\nWelcome to the Bird Park!\n\n"

bird1 = Duck.new "Donald"
bird2 = Penguin.new "Peter"
bird3 = Goose.new "Mother"
bird4 = Ostrich.new "Orville"
bird5 = Owl.new "Woodsey"
bird6 = Kiwi.new "Kool"

bird1.print
bird2.print
bird3.print
bird4.print
bird5.print
bird6.print

puts "\n\n"

File: cs214/projects/proj10/java/Birds.java
/* Birds.java illustrates inheritance and polymorphism.
 *
 * Begun by: Joel Adams, for CS 214 Lab 10 at Calvin College.
 * Completed by: Advait Scaria
 * Date: 04/23/2019s
 *********************************************************/

public class Birds {
  
  public static void main(String[] args) {
      System.out.println("\nWelcome to the Bird Park!\n");

      // Bird bird0 = new Bird();
      // Bird bird1 = new Bird("Hawkeye");
      Bird bird2 = new Duck("Donald");
      Bird bird3 = new Penguin("Peter");
      Bird bird4 = new Goose("Mother");
      Bird bird5 = new Ostrich("Orville");
      Bird bird6 = new Owl("Woodsey");
      Bird bird7 = new Kiwi("Kool");


      // bird1.print();
      bird2.print();
      bird3.print();
      bird4.print();
      bird5.print();
      bird6.print();
      bird7.print();
      System.out.println();
    }
}

